orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -818,3 +821,9 @@ def _update_onnx_model_initializers(self, state_tensors):
         for w_i in replace_indices:
             del self._onnx_model.graph.initializer[w_i]
         self._onnx_model.graph.initializer.extend(new_weights)
+
+    def _select_device_if_general_or_cpu_if_allfinite(self, device, name):
```

Comment:
I'd suggest a more intuitive name such as `_select_output_device` and add docstring explaining the behavior for picking `cpu` for `all_finite` and `device`, the arguments and return value.